File: 30-Days-of-Code/Day-6-Lets-Review/Haskell/solution.hs
import Control.Applicative
import Control.Monad
import Data.List

main = do
    num <- read <$> getLine :: IO Int
    replicateM num $ do
        s <- getLine
        let [firstHalf, secondHalf] = map reverse $ strSplit s [] [] 0
        let output = intercalate " " [firstHalf, secondHalf]
        putStrLn output

strSplit [] first second i = [first, second]
strSplit (x:xs) first second i = if i `mod` 2 == 0
                  then strSplit xs (x:first)    second  (i+1)
                  else strSplit xs     first (x:second) (i+1)

File: 30-Days-of-Code/Day-6-Lets-Review/Pascal/solution.pas
var n : int64;
    i,j : longint;
    s,firstHalf,secondHalf : ansistring;
    
begin
    readln(n);
    for i:=1 to n do
    begin
        readln(s);
        
        firstHalf := '';
        secondHalf := '';
        
        for j:=1 to length(s) do
        begin
            if j mod 2=0 then secondHalf:=secondHalf + s[j]
            else firstHalf := firstHalf + s[j];          
        end;
        writeln(firstHalf,' ',secondHalf);
    end;
end.
